foxpuppet/browser/window.py

```diff
@@ -3,22 +3,83 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from foxpuppet.windows import Windows
+from foxpuppet.browser.notifications import (
+    AddOnInstallBlocked,
+    AddOnInstallConfirmation,
+    AddOnInstallComplete,
+    AddOnInstallRestart,
+    AddOnInstallFailed,
+    AddOnProgress,
+    BaseNotification)
 
+from selenium.common.exceptions import NoSuchElementException
 from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
 
 
 class BrowserWindow(object):
     _file_menu_button_locator = (By.ID, 'file-menu')
     _file_menu_private_window_locator = (By.ID, 'menu_newPrivateWindow')
     _file_menu_new_window_button_locator = (By.ID, 'menu_newNavigator')
     _nav_bar_locator = (By.ID, 'nav-bar')
+    _notification_locator = (
+        By.CSS_SELECTOR, '#notification-popup popupnotification')
     _tab_browser_locator = (By.ID, 'tabbrowser-tabs')
 
     def __init__(self, selenium, *args, **kwargs):
         self.selenium = selenium
         self._windows = Windows(selenium)
 
     @property
+    def notification(self):
+        """Provides access to the currently displayed notification."""
+        _map = {
+            'addon-install-blocked-notification': AddOnInstallBlocked,
+            'addon-install-confirmation-notification':
+                AddOnInstallConfirmation,
+            'addon-install-complete-notification': AddOnInstallComplete,
+            'addon-install-restart-notification': AddOnInstallRestart,
+            'addon-install-failed-notification': AddOnInstallFailed,
+            'addon-progress-notification': AddOnProgress,
+        }
+
+        try:
+            with self.selenium.context('chrome'):
+                root = self.selenium.find_element(*self._notification_locator)
+                _id = root.get_attribute('id')
+                return _map.get(_id, BaseNotification)(self, root)
+        except NoSuchElementException:
+            return None  # no notification is displayed
+
+    def wait_for_notification(self, notification_class=BaseNotification,
+                              timeout=5):
```

Comment:
I think @jrbenny35 is working on this, so I can add a TODO to fix this up when I can simply use `self.wait` with the default timeout.